network-boot-lab.md

```diff
@@ -0,0 +1,282 @@
+      --p2p-listen-endpoint localhost:9010 \
+      --access-control-allow-origin=* \
+      --contracts-console \
+      --http-validate-host=false \
+      --verbose-http-errors \
+      --enable-stale-production \
+      --p2p-peer-address localhost:9011 \
+      --p2p-peer-address localhost:9012 \
+      --p2p-peer-address localhost:9013 \
+    >> $DATADIR"/nodeos.log" 2>&1 & \
+    echo $! > $DATADIR"/eosd.pid"
+    ```
+    > Remember to replace `EOS_PUB_DEV_KEY` and `EOS_PRIV_DEV_KEY` with the previous generated key pair.
+2. Finally, assign execution permission to the script:
+    ```bash
+    $ cd ~/biosboot/genesis/
+    $ chmod 755 genesis_start.sh
+    $ ./genesis_start.sh
+    ```
+Once the previous command were executed, the genesis node that will bring **eosio** as a name will be able to:
+1. Produce blocks
+1. Listen HTTP requests at `SPECIFIED_GENESIS_NODE_IP:8888`
+1. Listen connection requests with other nodes at `SPECIFIED_GENESIS_NODE_IP:9010`
+1. Print smart contracts output in the terminal
+
+## nodeos Service Registries
+You can inspect the nodeos service registries with the following commands:
+```bash
+$ cd ~/biosboot/genesis/
+$ tail -f ./blockchain/nodeos.log
+``` 
+> To exit log print mode, press CTRL+C.
+
+## `eosio.contracts` Configuration and Compilation
+It is necessary to install the latest version of `eosio.contracts` along with **v1.8.0** version. To facilitate the understanding of the steps here described, the latest version will be referred as `EOSIO_CONTRACTS_DIRECTORY` and the `v1.8.0` as `EOSIO_OLD_CONTRACTS_DIRECTORY`.
+
+The following installation steps must be followed in the order they appear.
+1. Clone and compile the `eosio.contracts`:
+    ```bash
+    $ cd ~
+    $ git clone https://github.com/EOSIO/eosio.contracts.git
+    $ cd ./eosio.contracts/
+    $ ./build.sh
+    $ cd ./build/contracts/
+    ```
+1. Clone and install `eosio.cdt` `v1.6.3` (necessary version to compile `eosio.contracts` `v1.8.0`):
+    ```bash
+    $ wget https://github.com/eosio/eosio.cdt/releases/download/v1.6.3/eosio.cdt_1.6.3-1-ubuntu-18.04_amd64.deb
+    $ sudo apt install ./eosio.cdt_1.6.3-1-ubuntu-18.04_amd64.deb
+    ```
+1. Clone and compile `eosio.contracts` `v1.8.0`:
+    ```bash
+    $ cd ~
+    $ git clone https://github.com/EOSIO/eosio.contracts.git eosio.contracts-1.8.x
+    $ cd ./eosio.contracts-1.8.x/
+    $ git checkout release/1.8.x
+    $ ./build.sh
+    $ cd ./build/contracts/
+    ```
+## `eosio.bios` Contract Deployment
+### Activate `PREACTIVATE_FEATURE`
+Due to updates introduced in v1.8 and v2.0, it is necessary to activate a protocol feature called PREACTIVATE_FEATURE. To activate this feature, execute the following command:
+```bash
+$ curl --request POST \
+	--url http://localhost:8888/v1/producer/schedule_protocol_feature_activations \
+	-d '{"protocol_features_to_activate": ["0ec7e080177b2c02b278d5088611686b49d739925a92d9bfcacd7fc6b74053bd"]}'
+```
+### Deploy **Old** Contracts Version
```

Comment:
is this still required in the latest version of eosio?

Reply:
Solved